O/openPMD_api/build_tarballs.jl

```diff
@@ -1,47 +1,93 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
+using Base.BinaryPlatforms
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "platforms", "mpi.jl"))
 
 name = "openPMD_api"
-version = v"0.14.3"
+version = v"0.15.0"
 
-julia_versions = [v"1.6.0", v"1.7.0", v"1.8.0"]
+# `v"1.6.3"` fails to build
+julia_versions = [v"1.7.0", v"1.8.0", v"1.9.0"]
 
 # Collection of sources required to complete build
 sources = [
     # ArchiveSource("https://github.com/openPMD/openPMD-api/archive/refs/tags/0.13.4.tar.gz",
     #               "46c013be5cda670f21969675ce839315d4f5ada0406a6546a91ec3441402cf5e"),
     # We temporarily use a feature branch instead of a released
     # version because the Julia bindings are not released yet
-    ArchiveSource("https://github.com/eschnett/openPMD-api/archive/32f4fe62bd92cad93c920a93a589211a95bd1543.tar.gz",
-                  "1c7f2b445ea58ca9cee40f4a31eb794889877b639c451f7c2bf8cb04db1538de"),
+    ArchiveSource("https://github.com/eschnett/openPMD-api/archive/84bce33ffe0f730ac13562a6f0ff142222efff95.tar.gz",
+                  "ab394d567946b34a0a163d02c6d237f2339cddaa29dce57716ae326d034da623"),
+    ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
```

Comment:
@eschnett Why do you need the MacOSX SDK?

Reply:
@giordano @staticfloat so the right answer is that @eschnett should patch openPMD to work with the old Mac SDK?

Reply:
Possibly.

Reply:
I don't recall why I need the new one. Maybe because I openPMD uses newer C++ features? In this case, using the older one would be difficult, that would be more than a simple patch.

Reply:
We need it for `std::filesystem`; there's a respective comment further down.